### PR TITLE
fix(python): preserve API state on failed operations

### PR DESCRIPTION
**Pull request by huanglune**
## Summary

- Avoid shared default `IndexParams` state between independently created `Index()` instances.
- Keep `Index` and `Collection` retryable after failed first-fit / first-insert paths instead of leaving half-initialized Python API state behind.
- Validate `Client` disk-delete preconditions before mutating in-memory maps, and normalize `Index.insert()` list inputs through the same numpy path as search APIs.
- Add focused regression coverage for each fixed API state edge case.

## Test plan

- [x] `git diff --check`
- [x] `timeout 60s .venv/bin/pytest -q`
- [x] `timeout 60s .venv/bin/ruff check python scripts tools`
- [x] pre-commit hooks during `git commit`

### CODE REVIEW

File: python/src/alayalite/index.py [hunk 1]
```diff
@@ -104,24 +104,30 @@ def fit(
...
             f"  vectors.shape: {vectors.shape}, num_threads: {num_threads}, ef_construction: {ef_construction}\n"
             f"start fitting index..."
         )
-        self.__index.fit(vectors, ef_construction, num_threads, item_ids, documents, metadata_list)
+        index = _PyIndexInterface(self.__params.to_cpp_params())
+        try:
+            index.fit(vectors, ef_construction, num_threads, item_ids, documents, metadata_list)
+        except Exception:
+            index.close_db()
+            raise
```
[the remark this answers is not in the document]
Comment (huanglune) [reply python/src/alayalite/index.py:164]: Fixed in 19019a0 by cleaning the configured RocksDB path after failed fit cleanup and preserving the original fit exception when close_db cleanup fails.

[the remark this answers is not in the document]
Comment (huanglune) [reply python/src/alayalite/index.py:164]: Follow-up in 183a8e5 limits removal to a RocksDB path that did not exist before fit, or was empty before fit, so failed cleanup does not delete pre-existing scalar storage.
